packages/medusa/src/migrations/1665660449659-add_analytics_config.ts

```diff
@@ -0,0 +1,18 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+import AnalyticsFeatureFlag from "../loaders/feature-flags/analytics"
+
+export const featureFlag = AnalyticsFeatureFlag.key
+
+export class addAnalyticsConfig1665660449659 implements MigrationInterface {
+  name = "addAnalyticsConfig1665660449659"
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `CREATE TABLE "analytics_config" ("id" character varying NOT NULL, "created_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "updated_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "deleted_at" TIMESTAMP WITH TIME ZONE, "user_id" character varying NOT NULL, "opt_out" boolean NOT NULL DEFAULT false, "anonymize" boolean NOT NULL DEFAULT false, CONSTRAINT "PK_ab46f7f4004ed899d568982c06c" PRIMARY KEY ("id", "user_id"))`
```

Comment:
todo: The primary key here will enforce uniqueness even if the config is deleted as the tuple still exists.
instead, I think that you should create a unique index id, user_id where deleted_at is null

Reply:
Something along the lines of this?

```suggestion
      await queryRunner.query(`CREATE TABLE "analytics_config" ("id" character varying NOT NULL, "created_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "updated_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "deleted_at" TIMESTAMP WITH TIME ZONE, "user_id" character varying NOT NULL, "opt_out" boolean NOT NULL DEFAULT false, "anonymize" boolean NOT NULL DEFAULT false, CONSTRAINT "PK_93505647c5d7cb479becb810b0f" PRIMARY KEY ("id"))`);
        await queryRunner.query(`CREATE UNIQUE INDEX "IDX_379ca70338ce9991f3affdeedf" ON "analytics_config" ("id", "user_id") WHERE deleted_at IS NULL`);
```

Reply:
yes exactly, I prefer to use human readable indeces names but the choice is yours. usually I compose the name like follow {constraint_type}_{table_name}_{columns}. but otherwise it is very nice ahah